Content.Shared/Silicons/StationAi/SharedStationAiSystem.cs

```diff
@@ -226,6 +238,41 @@ private void OnHolderInteract(Entity<StationAiHolderComponent> ent, ref AfterInt
         }
     }
 
+    private void OnHolderInteract(Entity<StationAiHolderComponent> ent, ref AfterInteractEvent args)
+    {
+        if (args.Handled || !args.CanReach || args.Target == null)
+            return;
+
+        if (!TryComp(args.Target, out StationAiHolderComponent? targetHolder))
+            return;
+
+        if (!TryComp(args.Used, out IntellicardComponent? intelliComp))
+            return;
+
+        var isUploading = _slots.CanEject(ent.Owner, args.User, ent.Comp.Slot);
+        var isDownloading = _slots.CanEject(args.Target.Value, args.User, targetHolder.Slot);
+
+        if (isUploading == isDownloading)
```

Comment:
So, if you try to upload an AI into a core that already has an AI in it, the operation fails?

There should be a pop up message to the uploader that you cannot upload an AI to a core that is already occupied